modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -334,11 +336,16 @@ public ChannelHandler configureServerChannelHandler() {
         return new HttpChannelHandler(this, handlingSettings);
     }
 
-    protected static final AttributeKey<Netty4HttpChannel> HTTP_CHANNEL_KEY = AttributeKey.newInstance("opensearch-http-channel");
+    public static final AttributeKey<Netty4HttpChannel> HTTP_CHANNEL_KEY = AttributeKey.newInstance("opensearch-http-channel");
```

Comment:
Can we keep this protected, and the other attribute keys protected, the only access of these values should be in the class that extends Netty4HttpServerTransport, if there are additional usages, they can be pass via the extending class to the usage.

Reply:
I needed to change the access modifier to refer to the attribute key in the companion PR (https://github.com/opensearch-project/security/pull/3418). The `header_verifier` is not a subclass of Netty4HttpServerTransport, but needed access. There are other ways of passing the info which can be looked into.

Reply:
I've removed this level of access for the moment, trying to streamline the changes in OpenSearch for the moment

Reply:
The change to this access modifier may not be needed after https://github.com/opensearch-project/security/pull/3430, but I had to keep it in for the current state of the header verifier in the security plugin.

Reply:
There should be only one instance of the AttributeKey. The AttributeKey is only usable in places where the netty channel context is present. 

In the security PR its used [here](https://github.com/opensearch-project/security/pull/3418/files#diff-62eb3270a6f8cc2b9e7b4a92393d1ed9c3ed291ba7548cfddfbe1140e715b0d1R90) which is not in the same package or a subclass so I changed the access modifier to get access to this constant. There are subclasses of this class in the security plugin where the header verifier is instantiated and I can pass it in through object construction, but it felt neater to change the access modifier to public.